test/core/tsi/s2a/s2a_test_util.cc

```diff
@@ -185,3 +187,220 @@ size_t expected_message_size(size_t plaintext_size) {
+    case TLS_AES_128_GCM_SHA256_ciphersuite: {
+      /**  This record arises from the plaintext 7891 when encrypted using the
+       *   AES-128-GCM cipher created by the create_example_session_state
+       *   method. In decimal form, the encrypted record is the following:
+       *   {23, 3, 3, 0, 21, 175, 119, 139, 14, 214, 110, 254, 231, 19, 251,
+       *   237, 162, 53, 123, 131, 47, 117, 149, 0, 238, 205}.
+       */
+      uint8_t correct_record_three_bytes[26] = {
+          0x17, 0x03, 0x03, 0x00, 0x15, 0xaf, 0x77, 0x8b, 0xe,
+          0xd6, 0x6e, 0xfe, 0xe7, 0x13, 0xfb, 0xed, 0xa2, 0x35,
+          0x7b, 0x83, 0x2f, 0x75, 0x95, 0x00, 0xee, 0xcd};
+      GPR_ASSERT(record_three_size == 26);
+      for (size_t i = 0; i < record_three_size; i++) {
+        GPR_ASSERT(record_three[i] == correct_record_three_bytes[i]);
+      }
+    } break;
+    case TLS_AES_256_GCM_SHA384_ciphersuite: {
+      /**  This record arises from the plaintext 7891 when encrypted using the
+       *   AES-256-GCM cipher created by the create_example_session_state
+       *   method. In decimal form, the encrypted record is the following:
+       *   {23, 3, 3, 0, 21, 150, 203, 132, 204, 93, 189, 39, 165, 19, 198,
+       *   210, 35, 204, 15, 74, 88, 64, 225, 126, 86, 216}.
+       */
+      uint8_t correct_record_three_bytes[26] = {
+          0x17, 0x03, 0x03, 0x00, 0x15, 0x96, 0xcb, 0x84, 0xcc,
+          0x5d, 0xbd, 0x27, 0xa5, 0x13, 0xc6, 0xd2, 0x23, 0xcc,
+          0x0f, 0x4a, 0x58, 0x40, 0xe1, 0x7e, 0x56, 0xd8};
+      GPR_ASSERT(record_three_size == 26);
+      for (size_t i = 0; i < record_three_size; i++) {
+        GPR_ASSERT(record_three[i] == correct_record_three_bytes[i]);
+      }
+    } break;
+    case TLS_CHACHA20_POLY1305_SHA256_ciphersuite: {
+      /**  This record arises from the plaintext 7891 when encrypted using the
+       *   CHACHA-POLY cipher created by the create_example_session_state
+       *   method. In decimal form, the encrypted record is the following:
+       *   {23, 3, 3, 0, 21, 146, 74, 83, 67, 207, 29, 198, 149, 9, 73, 65,
+       *   171, 161, 108, 108, 36, 48, 92, 200, 64, 138}.
+       */
+      uint8_t correct_record_three_bytes[26] = {
+          0x17, 0x03, 0x03, 0x00, 0x15, 0x92, 0x4a, 0x53, 0x43,
+          0xcf, 0x1d, 0xc6, 0x95, 0x09, 0x49, 0x41, 0xab, 0xa1,
+          0x6c, 0x6c, 0x24, 0x30, 0x5c, 0xc8, 0x40, 0x8a};
+      GPR_ASSERT(record_three_size == 26);
+      for (size_t i = 0; i < record_three_size; i++) {
+        GPR_ASSERT(record_three[i] == correct_record_three_bytes[i]);
+      }
+    } break;
+  }
+  return true;
+}
+
+bool check_record_empty_plaintext(TLSCiphersuite ciphersuite, uint8_t* record,
+                                  size_t record_size, char** error_details) {
+  if (record == nullptr && record_size != 0) {
+    *error_details =
+        gpr_strdup("If |record| is nullptr, then |record_size| must be zero.");
+    return false;
+  }
+  switch (ciphersuite) {
```

Comment:
We can use the function I mentioned above to test this logic as well.

Reply:
Done.